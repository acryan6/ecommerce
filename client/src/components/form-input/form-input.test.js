import { shallow } from "enzyme";
import React from "react";
import FormInput from "./form-input.component.jsx";

describe("formInput component", () => {
  let wrapper;
  let mockHandleChange;

  beforeEach(() => {
    mockHandleChange = jest.fn();

    const mockProps = {
      label: "email",
      value: "user@gmail.com",
      handleChange: mockHandleChange,
    };

    wrapper = shallow(<FormInput {...mockProps} />);
  });

  it("should render FormInput component", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should call handleChange when input changes", () => {
    wrapper.find("FormInputContainer").simulate("change");

    expect(mockHandleChange).toHaveBeenCalled();
  });

  it("should render FormInputLabel if there is a label", () => {
    expect(wrapper.exists("FormInputLabel")).toBe(true);
  });

  it("should not render FormInputLabel if there is no label", () => {
    const mockProps = {
      label: "",
      email: "user@gmail.com",
      handleChange: mockHandleChange,
    };

    const newWrapper = shallow(<FormInput {...mockProps} />);

    expect(newWrapper.exists("FormInputLabel")).toBe(false);
  });
});
