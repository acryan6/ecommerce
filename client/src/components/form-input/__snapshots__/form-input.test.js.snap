// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`formInput component should render FormInput component 1`] = `ShallowWrapper {}`;
